src/inputs/input/input.jsx

```diff
@@ -264,12 +267,19 @@ class Input extends React.PureComponent {
 
     onKeyDown(event) {
         const {
+            allowDecimals,
             onKeyDown,
         } = this.props;
 
+        const type = this.getType();
+
         if (isFunction(onKeyDown)) {
             onKeyDown(event);
         }
+
+        if (allowDecimals && type === 'number' && event.keyCode === DOT_KEY_CODE) {
+            event.preventDefault();
+        }
```

Comment:
Oh, just realized the existing logic is maybe not right?  If I am reading it right (bearing in mind I am doing this at 2:19 AM my time ... not sleeping well lately 😞 ) then this is doing `preventDefault()` on the `.` key if `allowDecimals` is `true`.  I don't think we want that.  I think we want something like this.
```JavaScript
if (type === 'number') { // don't even bother with anything further if it's not a number input
    if (event.keyCode === DOT_KEY_CODE && !allowDecimals) {
        event.preventDefault();
    }
}
```

If I am wrong and you tested your version and it is working as expected, my apologies! =)

Reply:
Let's linger about it on our scrum. May be terminology is not correct from my PR.